[protocol] Implement Get Host State

kesyog opened the pull request:
Add serializer, deserializer, and fuzz tests for Get Host State request
and response messages (6.22 in the Cerberus Protocol). This message is
used to query the host processor reset state.

Fixes #44

Signed-off-by: Kesavan Yogeswaran <hikes@google.com>

mcy commented on src/protocol/get_host_state.rs:
```diff
@@ -0,0 +1,129 @@
+make_fuzz_safe! {
+    /// The [`GetHostState`] response.
+    #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```
Uh oh, looks like `rustfmt` got confused. I bet you copy-pasted this from a place where I got this wrong, hehe.

kesyog replied:
Fixed. I had thought some macro parsing oddity earlier, but turns out I was misreading the compiler error I was fixing at the time.

mcy commented on src/protocol/mod.rs:
```diff
@@ -203,6 +210,10 @@ impl From<u8> for CommandType {
             0x02 => CommandType::DeviceCapabilities,
             0x03 => CommandType::DeviceId,
             0x04 => CommandType::DeviceInfo,
+            0x81 => CommandType::GetDigests,
+            0x82 => CommandType::GetCert,
+            0x83 => CommandType::Challenge,
```
Whoops.

Mind putting this in its own commit to the effect of "[protocol] Add missed entries to `ComandType::from()`"?

`git add -p` is your friend here. =)

kesyog replied:
Done

mcy reviewed (approved):
Two nits, but otherwise this is really awesome. I'm guessing this was mostly copy-and-pasted, which is fine, but I love to see that it's straightforward to implement protocol messages!

mcy commented:
Ah. It appears a recent refactor I did broke you, sorry. =P

Should be a matter of `s/(To/From)WireError/wire::Error/g`, plus or minus appropriate `use` statements.

kesyog commented:
Definitely was largely a copy-paste-rename job, but it helped to get a (little) better feel for the codebase.

I've rebased to include the latest wire error changes and fixed the feedback. @mcy looks like you have to re-approve the CI workflow for the latest push.

mcy commented:
LG. Will merge once CI finishes.
